go/arrow/array/numeric.gen.go

```diff
@@ -1085,6 +1377,22 @@ func (a *Duration) setData(data *Data) {
 	}
 }
 
+func (a *Duration) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+	return fmt.Sprint(time.Duration(a.values[i]) * a.DataType().(*arrow.DurationType).Unit.Multiplier())
```

Comment:
this logic is a little opaque to me, would you mind adding a comment on intended output.

Reply:
Are there overflow issues for courser precisions?

Reply:
Yes, there can end up being overflow issues for large enough values depending on the precision. I'll see if i can tweak this to try to avoid the overflows.

Reply:
fixed